configure.ac

```diff
@@ -145,6 +146,13 @@ AC_SEARCH_LIBS([gethostbyname], [nsl],,[AC_MSG_ERROR([cannot find gethostbyname(
 AC_SUBST([nsl_LIBS],[$LIBS])
 AX_RESTORE_FLAGS
 
+if test "x$enable_systemd_journal" = "xyes" ; then
+  AX_SAVE_FLAGS
+  AC_SEARCH_LIBS([sd_journal_send], [systemd-journal],,[AC_MSG_ERROR([cannot find sd_journal_send() function])])
```

Comment:
I would recommend using `PKG_CHECK_MODULES` (because it sets correctly not only the LIBS but also the CFLAGS.

Also (without deep knowledge about history) this code seems to work for CentOS 7, but Fedora 28+ seems not to have systemd-journal library at all and instead it's using libsystemd (https://www.freedesktop.org/software/systemd/man/sd-journal.html)